libs/coin-polkadot/src/logic.ts

```diff
@@ -24,11 +24,9 @@ export const isStash = (a: PolkadotAccount): boolean => {
 /**
  * Returns true if account is a controller.
  * when account is a controller, we have the information of which stash it controls
- *
- * @param {PolkadotAccount} a
  */
-export const isController = (a: PolkadotAccount): boolean => {
-  return !!a.polkadotResources?.stash;
+export const isStashAccount = (account: PolkadotAccount): boolean => {
+  return !!account.polkadotResources?.stash;
```

Comment:
resources && resources.stash != null ? I think it's equivalent and more readable

Btw can we add unit test on this ? Allowing us to check easily if both syntax are the same, test each branch eg:

- polkadotResources stash undefined/null
- polkadotResources stash defined
- polkadotResources hasn't a stash key defined

Reply:
If you don't update the method, i'll approve without unit tests there as we only cover what we edit
![image](https://github.com/LedgerHQ/ledger-live/assets/19901996/6160e12b-b65b-49e4-be90-ff6897f03755)

Reply:
Typechecking is not great here because stash is of type `string | null | undefined`
I can do the check `typeof account.polkadotResources?.stash !== undefined` which does what we want.

[source](https://stackoverflow.com/a/19891952/7280833)

Reply:
!= null includes both null and undefined, and the missing key

Reply:
I find `typeof account.polkadotResources?.stash !== undefined` even more weird than the base expression :sweat_smile:

Reply:
If a variable is not null doesn't mean it's not undefined right 🤔

Reply:
yes it means it's not undefined

!= null includes null, undefined, and key not present

Reply:
Ah yeah you mean using `!=` and not `!==` in that sense I don't think it's more clear than `!!` then

Reply:
Ok let's use the base condition then

My thoughts where :

We want resources to be defined (`resources`) and (`&&`) resource stash to be defined (`resources.stash != null `)

A bit more human